### PR TITLE
Merge `rc/3.4` into `main`

### PR DESCRIPTION
**Pull request by edoardopirovano**
Merges back the `rc/3.4` branch into `main`.

### CODE REVIEW

File: java/upgrades/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## 0.0.7
```
Comment (aeisenberg) [inline java/upgrades/CHANGELOG.md:1]: Same thing for the upgrades. Delete this folder.

**Review by aeisenberg — CHANGES_REQUESTED** [review]
Somehow there are some upgrades files that have been back ported. These should be removed. Also, some of the solorigate files as well.

**Comment by edoardopirovano** [comment]
Thanks @aeisenberg, looks like I incorrectly resolved the merge conflict with https://github.com/github/codeql/pull/7355 in favour of the RC branch instead of `main`. I've amended the merge commit to resolve in favour of `main` instead. Is it looking correct now?

**Review by aeisenberg — APPROVED** [review]
Looks good now.

**Pull request merged**